src/storage/zone_map_manager.cpp

```diff
@@ -0,0 +1,245 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+//
+//
+// Identification: src/storage/zone_map_manager.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "catalog/catalog.h"
+#include "catalog/zone_map_catalog.h"
+#include "catalog/database_catalog.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "storage/storage_manager.h"
+#include "storage/data_table.h"
+#include "type/ephemeral_pool.h"
+#include "storage/zone_map_manager.h"
+
+namespace peloton {
+namespace storage {
+
+// Get instance of the global zone map manager
+ZoneMapManager *ZoneMapManager::GetInstance() {
+  static ZoneMapManager global_zone_map_manager;
+  return &global_zone_map_manager;
+}
+
+ZoneMapManager::ZoneMapManager() {
+  zone_map_table_exists = false;
+  pool_.reset(new type::EphemeralPool());
+}
+
+/*@brief   The function creates the zone map table in catalog 
+*/
```

Comment:
Comment formatting in this file.

Reply:
Hey I didnt understand what you meant by comment formatting ?

Reply:
Use JavaDoc style in this file for comments.

Reply:
I dont know why this wasnt collapsed but I did use JavaDoc formatting in the new commit.